log10/load.py

```
@@ -68,67 +66,57 @@ def _func_with_backoff(func, *args, **kwargs):
 
 
-def get_session_id():
-    if target_service == "bigquery":
-        return str(uuid.uuid4())
-
+def post_request(url: str, json: dict = {}) -> requests.Response:
+    """
+    Authenticated POST request to log10.
+    """
+    json["organization_id"] = org_id
+    r = None
     try:
-        res = post_session_request()
-
-        return res.json()["sessionID"]
-    except requests.HTTPError as http_err:
+        # todo: set timeout
+        r = httpx_client.post(
+            url,
+            headers={"x-log10-token": token, "Content-Type": "application/json"},
+            json=json,
+        )
+        r.raise_for_status()
+    except httpx.HTTPError as http_err:
         if "401" in str(http_err):
-            raise Exception(
+            logging.error(
                 "Failed anthorization. Please verify that LOG10_TOKEN and LOG10_ORG_ID are set correctly and try again."
                 + "\nSee https://github.com/log10-io/log10#%EF%B8%8F-setup for details"
             )
         else:
-            raise Exception(f"Failed to create LOG10 session. Error: {http_err}")
-    except requests.ConnectionError:
-        raise Exception(
-            "Invalid LOG10_URL. Please verify that LOG10_URL is set correctly and try again."
-            + "\nSee https://github.com/log10-io/log10#%EF%B8%8F-setup for details"
-        )
+            logging.error(f"Failed to create LOG10 session. Error: {http_err}")
     except Exception as e:
-        raise Exception(
-            "Failed to create LOG10 session: "
-            + str(e)
-            + "\nLikely cause: LOG10 env vars missing or not picked up correctly!"
-            + "\nSee https://github.com/log10-io/log10#%EF%B8%8F-setup for details"
+        logger.error(f"LOG10: failed to insert in log10: {json} with error {e}")
+
+    return r
+
+
+def get_session_id():
+    """
+    Get session ID from log10.
+    """
+    res = post_request(url + "/api/sessions", {})
```

Since post_request swallows errors, we need to always check for None response eg `if not res return None`